mypy/checker.py

```diff
@@ -1530,7 +1530,13 @@ def bind_and_map_method(self, sym: SymbolTableNode, typ: FunctionLike,
         """
         if (isinstance(sym.node, (FuncDef, OverloadedFuncDef, Decorator))
                 and not is_static(sym.node)):
-            bound = bind_self(typ, self.scope.active_self_type())
+            if isinstance(sym.node, Decorator):
+                is_class_method = sym.node.func.is_class
+            elif isinstance(sym.node, OverloadedFuncDef):
+                is_class_method = sym.node.is_class
+            else:
+                is_class_method = False
```

Comment:
Should we check `is_class` for regular `FuncDef`s? (It can be implicit for certain functions, right?)

Reply:
Hm, you are right, it is better to be on the safe side.